cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -481,16 +481,14 @@ private NamespaceId getNamespaceId(byte[] rowKey) {
     void handle(NamespaceId namespaceId, Location location, String namespacedLogBaseDir);
   }
 
-  private static Location getCompatibleLocationFromValue(LocationFactory locationFactory,
-                                                         byte[] value) throws URISyntaxException {
+  private Location getCompatibleLocationFromValue(byte[] value) throws URISyntaxException {
     String locationPath = null;
-    URI locationURI;
     String valueString = Bytes.toString(value);
-    locationURI = new URI(valueString);
-    if (locationURI.getPath() == valueString) {
+    URI locationURI = new URI(valueString);
+    if (locationURI.getPath().equals(valueString)) {
```

Comment:
there is no need for everything below here, you can just create the location from `Locations.getRelativePath(rootLocationFactory, locationURI)`, assuming that method is fixed